backend/feed_service.py

```
@@ -27,12 +31,540 @@
+
+    # Initial progress announcement
+    announcer.announce(
+        msg=f"data: {json.dumps({'type': 'progress', 'status': 'Starting OPML import...', 'value': 0, 'max': 100})}\n\n"
+    )
+
+    total_feeds_to_import = 0
+    try:
+        # Use a temporary stream to count feeds without consuming the main stream
+        count_root = SafeET.fromstring(opml_content)
+        total_feeds_to_import = _count_feeds_in_opml(count_root)
+    except Exception:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)